napari/components/viewer_model.py

```diff
@@ -368,20 +368,18 @@ def _on_active_layer(self, event):
             self.camera.interactive = active_layer.interactive
 
     def _on_layers_change(self):
+        """Update the dimension and range of the dims property as well as the cursor position.
+
+        When the ranges are updated, we try to maintain the current slice plane as much
+        as possible.
         """
-        Update the dimension and range of the dims property as well as the cursor position.
-        When changing the dims property the currently selected view of the
-        viewer is tried to be maintained.
-        """
-        ranges = None
+        previous_point = None
         if len(self.layers) == 0:
             self.dims.ndim = 2
             self.dims.reset()
         else:
-            ranges = self.layers._ranges
```

Comment:
I rearranged this a little because I found it hard to keep track what was defined in the `else` block (i.e. `ranges` and `ndim`). This arrangement seems easier to follow and shouldn't introduce any performance concerns (since `LayerList._ranges` is cached).